### ISSUE TITLE
Error message for the tutorial

### ISSUE THREAD

**Issue by feiranl** [AlexanderKroll/KM_prediction_function#1]
Hi, I tried to use this tutorial, but there were some error message. I am wondering wether it is due to the package version, and may I know which version it should be for each package ? Thanks for help!

Below is the code I run and the error message.

At first, I met a error message when I run this:
```
import esm
model, alphabet = torch.hub.load("facebookresearch/esm:main", "esm1b_t33_650M_UR50S")
```
```
Using cache found in /Users/feiranl/.cache/torch/hub/facebookresearch_esm_main
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/ipykernel_52431/198034724.py in <module>
      1 import esm
----> 2 model, alphabet = torch.hub.load("facebookresearch/esm:main", "esm1b_t33_650M_UR50S")

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/torch/hub.py in load(repo_or_dir, model, source, force_reload, verbose, skip_validation, *args, **kwargs)
    402         repo_or_dir = _get_cache_or_reload(repo_or_dir, force_reload, verbose, skip_validation)
    403 
--> 404     model = _load_local(repo_or_dir, model, *args, **kwargs)
    405     return model
    406 

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/torch/hub.py in _load_local(hubconf_dir, model, *args, **kwargs)
    428 
    429     hubconf_path = os.path.join(hubconf_dir, MODULE_HUBCONF)
--> 430     hub_module = _import_module(MODULE_HUBCONF, hubconf_path)
    431 
    432     entry = _load_entry_from_hubconf(hub_module, model)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/torch/hub.py in _import_module(name, path)
     74     module = importlib.util.module_from_spec(spec)
     75     assert isinstance(spec.loader, Loader)
---> 76     spec.loader.exec_module(module)
     77     return module
     78 

/opt/anaconda3/envs/deepKM/lib/python3.7/importlib/_bootstrap_external.py in exec_module(self, module)

/opt/anaconda3/envs/deepKM/lib/python3.7/importlib/_bootstrap.py in _call_with_frames_removed(f, *args, **kwds)

~/.cache/torch/hub/facebookresearch_esm_main/hubconf.py in <module>
      6 dependencies = ["torch"]
      7 
----> 8 from esm.pretrained import (
      9     esm1_t34_670M_UR50S,
     10     esm1_t34_670M_UR50D,

ImportError: cannot import name 'esm_if1_gvp4_t16_142M_UR50' from 'esm.pretrained' (/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/esm/pretrained.py)

```
After I looked into your code and used another way to manage load model, but then I got another error message when I run the following code:
```
KM_predicton(substrate_list = ["InChI=1S/C21H27N7O14P2/c22-17-12-19(25-7-24-17)28(8-26-12)21-16(32)14(30)11(41-21)6-39-44(36,37)42-43(34,35)38-5-10-13(29)15(31)20(40-10)27-3-1-2-9(4-27)18(23)33/h1-4,7-8,10-11,13-16,20-21,29-32H,5-6H2,(H5-,22,23,24,25,33,34,35,36,37)/p-1/t10-,11-,13-,14-,15-,16-,20-,21-/m1/s1"], 
             enzyme_list = ["MSIPETQKGVIFYESHGKLEYKDIPVPKPKANELLINVKYSGVCHTDLHAWHGDWPLPVKLPLVGGHEGAGVVVGMGENVKWKIGDYAGIKWLNGSCMACEYCELGNESNCPHADLSGYTHDGSFQQYATADAVQAAHIPQGTDLAQVAPILCAGITVYKALKSANLMAGHWVAISGAAGGLGSLAVQYAKAMGYRVLGIDGGEGKEELFRSIGGEVFIDFTKEKDIVGAVLKATDGGAHGVINVSVSEAAIEASTRYVRANGTTVLVGMPAGAKCCSDVFNQVVKSISIVGSYVGNRADTREALDFFARGLVKSPIKVVGLSTLPEIYEKMEKGQIVGRYVVDTSK"])

```

```
Step 1/3: Calculating numerical representations for all metabolites.
.....1(a) Calculating input matrices for Graph Neural Network
.....1(b) Calculating numerical metabolite representations using a Graph Neural Network
2022-05-16 00:13:37.762152: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Step 2/3: Calculating numerical representations for all enzymes.
.....2(a) Loading ESM-1b model.
.....2(b) Calculating enzyme representations.
Step 3/3: Making predictions for KM.
---------------------------------------------------------------------------
XGBoostError                              Traceback (most recent call last)
/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/ipykernel_50215/2579094829.py in <module>
      2 
      3 KM_predicton(substrate_list = ["InChI=1S/C21H27N7O14P2/c22-17-12-19(25-7-24-17)28(8-26-12)21-16(32)14(30)11(41-21)6-39-44(36,37)42-43(34,35)38-5-10-13(29)15(31)20(40-10)27-3-1-2-9(4-27)18(23)33/h1-4,7-8,10-11,13-16,20-21,29-32H,5-6H2,(H5-,22,23,24,25,33,34,35,36,37)/p-1/t10-,11-,13-,14-,15-,16-,20-,21-/m1/s1"], 
----> 4              enzyme_list = ["MSIPETQKGVIFYESHGKLEYKDIPVPKPKANELLINVKYSGVCHTDLHAWHGDWPLPVKLPLVGGHEGAGVVVGMGENVKWKIGDYAGIKWLNGSCMACEYCELGNESNCPHADLSGYTHDGSFQQYATADAVQAAHIPQGTDLAQVAPILCAGITVYKALKSANLMAGHWVAISGAAGGLGSLAVQYAKAMGYRVLGIDGGEGKEELFRSIGGEVFIDFTKEKDIVGAVLKATDGGAHGVINVSVSEAAIEASTRYVRANGTTVLVGMPAGAKCCSDVFNQVVKSISIVGSYVGNRADTREALDFFARGLVKSPIKVVGLSTLPEIYEKMEKGQIVGRYVVDTSK"])
      5 

~/Documents/GitHub/KM_prediction_function/code/KM_prediction.py in KM_predicton(substrate_list, enzyme_list)
     35     df_KM_valid.reset_index(inplace = True, drop = True)
     36     X = calculate_xgb_input_matrix(df = df_KM_valid)
---> 37     KMs = predict_KM(X)
     38 
     39     df_KM_valid["KM [mM]"] = KMs

~/Documents/GitHub/KM_prediction_function/code/KM_prediction.py in predict_KM(X)
     49 
     50 def predict_KM(X):
---> 51         bst = pickle.load(open(join(CURRENT_DIR, "..", "data", "saved_models", "xgboost", "xgboost_model_new_KM_esm1b.dat"), "rb"))
     52         dX = xgb.DMatrix(X)
     53         KMs = 10**bst.predict(dX)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in __setstate__(self, state)
   1091             ptr = (ctypes.c_char * len(buf)).from_buffer(buf)
   1092             _check_call(
-> 1093                 _LIB.XGBoosterUnserializeFromBuffer(handle, ptr, length))
   1094             state['handle'] = handle
   1095         self.__dict__.update(state)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in _check_call(ret)
    187     """
    188     if ret != 0:
--> 189         raise XGBoostError(py_str(_LIB.XGBGetLastError()))
    190 
    191 

XGBoostError: [00:14:59] /private/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/pip-install-o_39px81/xgboost_bd82b85c21b54d8a876c4e8840e61141/xgboost/include/xgboost/json.h:65: Invalid cast, from Null to Array
Stack trace:
  [bt] (0) 1   libxgboost.dylib                    0x0000000107814ab0 dmlc::LogMessageFatal::~LogMessageFatal() + 112
  [bt] (1) 2   libxgboost.dylib                    0x0000000107842475 xgboost::JsonArray const* xgboost::Cast<xgboost::JsonArray const, xgboost::Value>(xgboost::Value*) + 549
  [bt] (2) 3   libxgboost.dylib                    0x00000001078fa91a xgboost::RegTree::LoadModel(xgboost::Json const&) + 1402
  [bt] (3) 4   libxgboost.dylib                    0x000000010789876c xgboost::gbm::GBTreeModel::LoadModel(xgboost::Json const&) + 492
  [bt] (4) 5   libxgboost.dylib                    0x0000000107889358 xgboost::gbm::GBTree::LoadModel(xgboost::Json const&) + 376
  [bt] (5) 6   libxgboost.dylib                    0x000000010789ab1c xgboost::LearnerImpl::LoadModel(xgboost::Json const&) + 1612
  [bt] (6) 7   libxgboost.dylib                    0x00000001078a201d xgboost::LearnerImpl::Load(dmlc::Stream*) + 557
  [bt] (7) 8   libxgboost.dylib                    0x0000000107811508 XGBoosterUnserializeFromBuffer + 168
  [bt] (8) 9   libffi.8.dylib                      0x00000001006d0d92 ffi_call_unix64 + 82
```
This is the line that reports the error:
`bst = pickle.load(open(join(CURRENT_DIR, "..", "data", "saved_models", "xgboost", "xgboost_model_new_KM_esm1b.dat"), "rb"))`

```
---------------------------------------------------------------------------
XGBoostError                              Traceback (most recent call last)
/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/ipykernel_50215/51671085.py in <module>
----> 1 bst = pickle.load(open(join(CURRENT_DIR, "..", "data", "saved_models", "xgboost", "xgboost_model_new_KM_esm1b.dat"), "rb"))

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in __setstate__(self, state)
   1091             ptr = (ctypes.c_char * len(buf)).from_buffer(buf)
   1092             _check_call(
-> 1093                 _LIB.XGBoosterUnserializeFromBuffer(handle, ptr, length))
   1094             state['handle'] = handle
   1095         self.__dict__.update(state)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in _check_call(ret)
    187     """
    188     if ret != 0:
--> 189         raise XGBoostError(py_str(_LIB.XGBGetLastError()))
    190 
    191 

XGBoostError: [00:16:28] /private/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/pip-install-o_39px81/xgboost_bd82b85c21b54d8a876c4e8840e61141/xgboost/include/xgboost/json.h:65: Invalid cast, from Null to Array
Stack trace:
  [bt] (0) 1   libxgboost.dylib                    0x0000000107814ab0 dmlc::LogMessageFatal::~LogMessageFatal() + 112
  [bt] (1) 2   libxgboost.dylib                    0x0000000107842475 xgboost::JsonArray const* xgboost::Cast<xgboost::JsonArray const, xgboost::Value>(xgboost::Value*) + 549
  [bt] (2) 3   libxgboost.dylib                    0x00000001078fa91a xgboost::RegTree::LoadModel(xgboost::Json const&) + 1402
  [bt] (3) 4   libxgboost.dylib                    0x000000010789876c xgboost::gbm::GBTreeModel::LoadModel(xgboost::Json const&) + 492
  [bt] (4) 5   libxgboost.dylib                    0x0000000107889358 xgboost::gbm::GBTree::LoadModel(xgboost::Json const&) + 376
  [bt] (5) 6   libxgboost.dylib                    0x000000010789ab1c xgboost::LearnerImpl::LoadModel(xgboost::Json const&) + 1612
  [bt] (6) 7   libxgboost.dylib                    0x00000001078a201d xgboost::LearnerImpl::Load(dmlc::Stream*) + 557
  [bt] (7) 8   libxgboost.dylib                    0x0000000107811508 XGBoosterUnserializeFromBuffer + 168
  [bt] (8) 9   libffi.8.dylib                      0x00000001006d0d92 ffi_call_unix64 + 82

```

**Comment by AlexanderKroll** [comment]
Hi, 

it seems that the current version of the fair-esm package does not support the download of the esm1b model in the say way as before.

You need to replace
>model, alphabet = torch.hub.load("facebookresearch/esm:main", "esm1b_t33_650M_UR50S")

with

>model, alphabet = torch.hub.load("facebookresearch/esm:v0.4.0", "esm1b_t33_650M_UR50S")

Below, I am listing the versions of the python packages that I used. I also added them to the Readme page.

- python 3.7.0
- tensorflow 2.3.1
- pandas 1.1.3
- torch 1.7.1
- rdkit 2020.09.1
- fair-esm 0.3.1
- py-xgboost 1.3.1
- pickle 4.0

Best,
Alex

**Comment by feiranl** [comment]
Thanks for your reply! After I made the changes accordingly, the downloading section is successful, but I still got error message in the prediction part as below:

```
Step 1/3: Calculating numerical representations for all metabolites.
.....1(a) Calculating input matrices for Graph Neural Network
.....1(b) Calculating numerical metabolite representations using a Graph Neural Network

2022-05-28 16:32:35.128992: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.

Step 2/3: Calculating numerical representations for all enzymes.
.....2(a) Loading ESM-1b model.
.....2(b) Calculating enzyme representations.
Step 3/3: Making predictions for KM.

---------------------------------------------------------------------------
XGBoostError                              Traceback (most recent call last)
/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/ipykernel_1281/2579094829.py in <module>
      2 
      3 KM_predicton(substrate_list = ["InChI=1S/C21H27N7O14P2/c22-17-12-19(25-7-24-17)28(8-26-12)21-16(32)14(30)11(41-21)6-39-44(36,37)42-43(34,35)38-5-10-13(29)15(31)20(40-10)27-3-1-2-9(4-27)18(23)33/h1-4,7-8,10-11,13-16,20-21,29-32H,5-6H2,(H5-,22,23,24,25,33,34,35,36,37)/p-1/t10-,11-,13-,14-,15-,16-,20-,21-/m1/s1"], 
----> 4              enzyme_list = ["MSIPETQKGVIFYESHGKLEYKDIPVPKPKANELLINVKYSGVCHTDLHAWHGDWPLPVKLPLVGGHEGAGVVVGMGENVKWKIGDYAGIKWLNGSCMACEYCELGNESNCPHADLSGYTHDGSFQQYATADAVQAAHIPQGTDLAQVAPILCAGITVYKALKSANLMAGHWVAISGAAGGLGSLAVQYAKAMGYRVLGIDGGEGKEELFRSIGGEVFIDFTKEKDIVGAVLKATDGGAHGVINVSVSEAAIEASTRYVRANGTTVLVGMPAGAKCCSDVFNQVVKSISIVGSYVGNRADTREALDFFARGLVKSPIKVVGLSTLPEIYEKMEKGQIVGRYVVDTSK"])
      5 

~/Documents/GitHub/KM_prediction_function/code/KM_prediction.py in KM_predicton(substrate_list, enzyme_list)
     35     df_KM_valid.reset_index(inplace = True, drop = True)
     36     X = calculate_xgb_input_matrix(df = df_KM_valid)
---> 37     KMs = predict_KM(X)
     38 
     39     df_KM_valid["KM [mM]"] = KMs

~/Documents/GitHub/KM_prediction_function/code/KM_prediction.py in predict_KM(X)
     49 
     50 def predict_KM(X):
---> 51         bst = pickle.load(open(join(CURRENT_DIR, "..", "data", "saved_models", "xgboost", "xgboost_model_new_KM_esm1b.dat"), "rb"))
     52         dX = xgb.DMatrix(X)
     53         KMs = 10**bst.predict(dX)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in __setstate__(self, state)
   1091             ptr = (ctypes.c_char * len(buf)).from_buffer(buf)
   1092             _check_call(
-> 1093                 _LIB.XGBoosterUnserializeFromBuffer(handle, ptr, length))
   1094             state['handle'] = handle
   1095         self.__dict__.update(state)

/opt/anaconda3/envs/deepKM/lib/python3.7/site-packages/xgboost/core.py in _check_call(ret)
    187     """
    188     if ret != 0:
--> 189         raise XGBoostError(py_str(_LIB.XGBGetLastError()))
    190 
    191 

XGBoostError: [16:33:22] /private/var/folders/h_/wl_qr2l56m380tydjnvnb9541bkd_z/T/pip-install-o_39px81/xgboost_bd82b85c21b54d8a876c4e8840e61141/xgboost/include/xgboost/json.h:65: Invalid cast, from Null to Array
Stack trace:
  [bt] (0) 1   libxgboost.dylib                    0x000000011a776ab0 dmlc::LogMessageFatal::~LogMessageFatal() + 112
  [bt] (1) 2   libxgboost.dylib                    0x000000011a7a4475 xgboost::JsonArray const* xgboost::Cast<xgboost::JsonArray const, xgboost::Value>(xgboost::Value*) + 549
  [bt] (2) 3   libxgboost.dylib                    0x000000011a85c91a xgboost::RegTree::LoadModel(xgboost::Json const&) + 1402
  [bt] (3) 4   libxgboost.dylib                    0x000000011a7fa76c xgboost::gbm::GBTreeModel::LoadModel(xgboost::Json const&) + 492
  [bt] (4) 5   libxgboost.dylib                    0x000000011a7eb358 xgboost::gbm::GBTree::LoadModel(xgboost::Json const&) + 376
  [bt] (5) 6   libxgboost.dylib                    0x000000011a7fcb1c xgboost::LearnerImpl::LoadModel(xgboost::Json const&) + 1612
  [bt] (6) 7   libxgboost.dylib                    0x000000011a80401d xgboost::LearnerImpl::Load(dmlc::Stream*) + 557
  [bt] (7) 8   libxgboost.dylib                    0x000000011a773508 XGBoosterUnserializeFromBuffer + 168
  [bt] (8) 9   libffi.8.dylib                      0x000000010959fd92 ffi_call_unix64 + 82
```

**Comment by feiranl** [comment]
Solved by install xgboost=1.3.3 using `conda install -c conda-forge py-xgboost=1.3.3` since I cannot find py-xgboost 1.3.1
